### PR TITLE
Add clang sanitizers

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -33,10 +33,17 @@
           crypto3-debug-tests = (pkgs.callPackage ./crypto3.nix {
             enableDebug = true;
             runTests = true;
+            sanitize = true;
```
Comment (akokoshn) [inline flake.nix:36]: why sanitize turned on for just debug test configuration?

File: flake.nix [hunk 2]
```diff
@@ -33,10 +33,17 @@
...
           });
           crypto3-sanitize = (pkgs.callPackage ./crypto3.nix {
-            enableDebug = true;
             runTests = true;
+            enableDebug = false;
+            sanitize = true;
+          });
+          crypto3-clang-sanitize = (pkgs.callPackage ./crypto3.nix {
```
Comment (akokoshn) [inline flake.nix:43]: maybe just `crypto3-clang-sanitize` enough?

File: flake.nix [hunk 3]
```diff
@@ -131,14 +143,20 @@
...
+            runTests = true;
+            enableDebug = false;
+            sanitize = true;
+          });
 
           all-clang = pkgs.symlinkJoin {
             name = "all";
             paths = [ crypto3-clang parallel-crypto3-clang proof-producer-clang ];
           };
           all-sanitizers = pkgs.symlinkJoin {
```
Comment (akokoshn) [inline flake.nix:157]: all-clang-sanitize

File: parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp [hunk 4]
```diff
@@ -55,7 +55,8 @@ BOOST_AUTO_TEST_CASE(vector_multiplication_test) {
         }, nil::crypto3::ThreadPool::PoolLevel::HIGH));
 
     for (std::size_t i = 0; i < size; ++i) {
-        BOOST_CHECK(v[i] == i * i);
+        if(v[i] != i * i) // If uncomment, test result parser will log an error: CData section too big found
```
Comment (akokoshn) [inline parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp:58]: looks strange.
Condition should be checked in BOOST_CHECK

Comment (AndreyMlashkin) [reply parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp:58]: It cause error on the parser side:
https://github.com/NilFoundation/placeholder/actions/runs/12015936613/job/33495224928

File: flake.nix (hunk 1 above)
Comment (AndreyMlashkin) [reply flake.nix]: Reverted

File: flake.nix (hunk 2 above)
Comment (AndreyMlashkin) [reply flake.nix:42]: You mean, we don't need to check
parallel-crypto3-clang-sanitize and proof-producer-sanitize ? 

Why?

File: parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp (hunk 4 above)
Comment (akokoshn) [reply parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp:58]: `BOOST_CHECK_EQUAL` helps?

File: flake.nix (hunk 2 above)
Comment (AndreyMlashkin) [reply flake.nix:37]: corrected

File: parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp (hunk 4 above)
Comment (AndreyMlashkin) [reply parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp]: trying now

File: flake.nix [hunk 5]
```diff
@@ -131,14 +142,20 @@
             runTests = true;
             enableDebug = false;
           });
+          proof-producer-sanitize = (pkgs.callPackage ./proof-producer.nix {
```
[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply flake.nix]: renamed

File: parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp (hunk 4 above)
Comment (AndreyMlashkin) [reply parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp]: @akokoshn  that didn't help
https://github.com/NilFoundation/placeholder/actions/runs/12031864193/job/33545980584

Comment (AndreyMlashkin) [reply parallel-crypto3/libs/parallelization-utils/test/thread_pool.cpp]: I fixed it a different way

File: flake.nix [hunk 6]
```diff
@@ -34,11 +34,24 @@
...
           });
-          crypto3-sanitize = (pkgs.callPackage ./crypto3.nix {
-            enableDebug = true;
+          crypto3-clang-sanitize = (pkgs.callPackage ./crypto3.nix {
+            stdenv = pkgs.llvmPackages_19.stdenv;
             runTests = true;
+            enableDebug = false;
             sanitize = true;
           });
+          crypto3-clang-sanitize-bench = (pkgs.callPackage ./crypto3.nix {
```
Comment (akokoshn) [inline flake.nix:43]: why we need sanitize and bench at the same time?

File: flake.nix [hunk 7]
```diff
@@ -111,6 +129,13 @@
             enableDebug = false;
             sanitize = true;
           });
+          crypto3-clang-sanitize-bench = (pkgs.callPackage ./crypto3.nix {
```
Comment (akokoshn) [inline flake.nix:132]: looks like duplicate

**Review by akokoshn — COMMENTED** [review]
Let's enable:
`crypto3-clang-sanitize`
`crypto3-clang-bench`
`crypto3-gcc-bench`
`parallel-crypto3-clang-sanitize`
`parallel-crypto3-clang-bench`
`parallel-crypto3-gcc-bench`

File: flake.nix (hunk 6 above)
Comment (AndreyMlashkin) [reply flake.nix]: Removed

File: flake.nix (hunk 7 above)
Comment (AndreyMlashkin) [reply flake.nix]: removed

File: crypto3/benchmarks/CMakeLists.txt [hunk 8]
```diff
@@ -60,7 +60,7 @@ set(BENCHMARK_NAMES
 
     "multiprecision/modular_adaptor_fixed"
 
-    "zk/lpc"
+    #"zk/lpc"
```
[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply crypto3/benchmarks/CMakeLists.txt:63]: It's failing now

[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply crypto3/benchmarks/CMakeLists.txt]: enabled

**Pull request merged**